In [1]:
pip install numpy

In [2]:
import numpy as n
arr=n.zeros((10,3))
arr

array([[0., 0., 0.],
       [0., 0., 0.],
       [0., 0., 0.],
       [0., 0., 0.],
       [0., 0., 0.],
       [0., 0., 0.],
       [0., 0., 0.],
       [0., 0., 0.],
       [0., 0., 0.],
       [0., 0., 0.]])

In [3]:
import numpy as np
arr=np.ones((5,7))
print(arr)

[[1. 1. 1. 1. 1. 1. 1.]
 [1. 1. 1. 1. 1. 1. 1.]
 [1. 1. 1. 1. 1. 1. 1.]
 [1. 1. 1. 1. 1. 1. 1.]
 [1. 1. 1. 1. 1. 1. 1.]]


In [5]:
import numpy as np
arr=np.full((4,7),9)
print(arr)

[[9 9 9 9 9 9 9]
 [9 9 9 9 9 9 9]
 [9 9 9 9 9 9 9]
 [9 9 9 9 9 9 9]]


In [7]:
import numpy as np
arr = np.empty((5,5))
print(arr)


[[5.19712638e-315 0.00000000e+000 0.00000000e+000 1.30532144e-320
  0.00000000e+000]
 [1.20662612e-186 5.84903538e-303 1.54611945e-229 1.10412726e-315
  0.00000000e+000]
 [5.41056497e-230 1.52666285e-321 2.35329361e-310 0.00000000e+000
  0.00000000e+000]
 [4.31198067e-298 1.06282428e-313 4.31198067e-298 1.06282426e-313
  4.31198067e-298]
 [1.06282422e-313 3.48740885e-316 5.58042878e-308 0.00000000e+000
  1.81816158e-321]]


In [8]:
import numpy as np
arr=np.eye(5)
print(arr)

[[1. 0. 0. 0. 0.]
 [0. 1. 0. 0. 0.]
 [0. 0. 1. 0. 0.]
 [0. 0. 0. 1. 0.]
 [0. 0. 0. 0. 1.]]


In [9]:
import numpy as np
Even = np.arange(2,200,2)
print(Even)


[  2   4   6   8  10  12  14  16  18  20  22  24  26  28  30  32  34  36
  38  40  42  44  46  48  50  52  54  56  58  60  62  64  66  68  70  72
  74  76  78  80  82  84  86  88  90  92  94  96  98 100 102 104 106 108
 110 112 114 116 118 120 122 124 126 128 130 132 134 136 138 140 142 144
 146 148 150 152 154 156 158 160 162 164 166 168 170 172 174 176 178 180
 182 184 186 188 190 192 194 196 198]


In [15]:
import numpy as np
arr = np.linspace(1,20,7)
print(arr)

[ 1.          4.16666667  7.33333333 10.5        13.66666667 16.83333333
 20.        ]


In [14]:
import numpy as np
arr = np.arange(1,100)
print(arr)

[ 1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24
 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48
 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72
 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 96
 97 98 99]


In [16]:
import numpy as np
arr = np.random.rand(10)
print(arr)

[0.96767838 0.01658053 0.80162786 0.46226001 0.66457086 0.06466437
 0.42204876 0.32050364 0.47208215 0.58063527]


rand is a funtion its only getting postive values between 0 to 120

In [17]:
import numpy as np
arr = np.random.randn(10)
print(arr)

[-0.48987303  0.16007565  1.13182785 -0.10845946  0.30966304 -0.20739
  0.33349957  0.54350379  0.94006009 -0.00556488]


In [21]:
import numpy as np
arr = np.random.randint(20)
print(arr)

16


randint is run in differnt output


In [25]:
import numpy as np
colours = ["red","green","pink","orange","blue","white","black"]
print(np.random.choice(colours,2))

['white' 'orange']


In [27]:
import numpy as np
arr = np.array([1,2,3])
new = arr.copy()
new[1] = 100
print("Array value is:",arr)
print("copy value is:",new)

Array value is: [1 2 3]
copy value is: [  1 100   3]


In [28]:
# view()
import numpy as np
arr = np.array([1,2,3])
new = arr.view()
new[1]=100
print("Array value is:",arr)
print("view value is:",new)

Array value is: [  1 100   3]
view value is: [  1 100   3]


In [30]:
import numpy as np
arr=np.array([5,10,15,20,25,30])
arr += 10
print("Update array value is:",arr)

Update array value is: [15 20 25 30 35 40]


In [31]:
import numpy as np
A = np.array([1,2,3,4,5])
B = np.array([6,7,8,9,10])
print(np.add(A,B))

[ 7  9 11 13 15]


In [32]:
import numpy as np
A = np.array([1,2,3,4,5])
B = np.array([6,7,8,9,10])
print(np.multiply(A,B))

[ 6 14 24 36 50]


In [33]:
import numpy as np
A = np.array([1,2,3,4,5])
B = np.array([6,7,8,9,10])
print(np.subtract(A,B))

[-5 -5 -5 -5 -5]


In [34]:
import numpy as np
A = np.array([1,2,3,4,5])
B = np.array([6,7,8,9,10])
print(np.divide(A,B))

[0.16666667 0.28571429 0.375      0.44444444 0.5       ]


In [35]:
import numpy as np
A = np.array([1,2,3,4,5])
B = np.array([6,7,8,9,10])
print(np.dot(A,B))

130


In [36]:
import numpy as np
Arr = np.array([1,2,3,4,5,6,7,8,9])
print(sum(Arr))

45


In [37]:
import numpy as np
Arr = np.array([1,2,3,4,5,6,7,8,9])
print(min(Arr))

1


In [42]:
import numpy as np
Arr = np.array([1,2,3,4,5,6,7,8,9])
print(np.std(Arr))

2.581988897471611


In [43]:
import numpy as np
Arr = np.array([1,2,3,4,5,6,7,8,9])
print(np.sqrt(Arr))

[1.         1.41421356 1.73205081 2.         2.23606798 2.44948974
 2.64575131 2.82842712 3.        ]


In [44]:
import numpy as np
Arr = np.array([1,2,3,4,5,6,7,8,9])
print(np.max(Arr))

9


In [45]:
import numpy as np
Arr = np.array([1,2,3,4,5,6,7,8,9])
print(np.mean(Arr))

5.0


In [46]:
import numpy as np
Arr = np.array([1,2,3,4,5,6,7,8,9])
print(np.sort(Arr))

[1 2 3 4 5 6 7 8 9]


In [47]:
import numpy as np
Arr = np.array([1,2,3,4,5,6,7,8,9])
print(np.unique(Arr))

[1 2 3 4 5 6 7 8 9]


In [50]:
imimport numpy as np
Arr = np.array([1,2,3,4,5,6,7,8,9])
print(np.sqrt(Arr))

[1 2 3 4 5 6 7 8]


In [51]:
import numpy as np
Arr = np.array([1,2,3,4,5,6,7,8,9])
print(np.sqrt(Arr))

(13,)
13


In [52]:
import numpy as np
arr=np.array([1,2,3,4,5,6,7,8,9])
print(arr.shape)
print(arr.size)
print(arr.itemsize)
print(arr.ndim)
print(arr.dtype)

(9,)
9
8
1
int64


In [54]:
import numpy as np
arr=np.array([[1,2,3,4],[5,4,3,2]])
print(arr.shape)

(2, 4)


In [56]:
import numpy as np
arr=np.array([[1,2,3,4],[5,4,3,2]])
print(arr.shape)
print(arr.ndim)
print(arr[0,1])
print(arr[1,1])
print(arr[:,1])

(2, 4)
2
2
4
[2 4]


In [57]:
 import numpy as np
 arr=np.array([1,2,3,4,5,6,7,8,99,10,11,12,13,14,15])
 print("original array",arr)
 new=arr.reshape(5,3)
 print(new)

original array [ 1  2  3  4  5  6  7  8 99 10 11 12 13 14 15]
[[ 1  2  3]
 [ 4  5  6]
 [ 7  8 99]
 [10 11 12]
 [13 14 15]]
